lessons/module-2/test-driven-development.md

```diff
@@ -345,6 +314,91 @@ Box
 1 passing (10ms)  
 ```
 
+## Assertion Library: Chai  
+
+An assertion is the crucial piece of the test that actually checks that when certain pieces of are code are executed, what we're getting back is what we expect.  
+
+In english, here are a few examples of things we might want to assert:  
+
+"After I create a new instance of a unicorn, the unicorn **should have a name**"  
+"After I create a new instance of a unicorn, and tell it to sing, the unicorn **should return a string of text from a song**"  
+"After I create a new instance of a unicorn, and the unicorn eats three times, the unicorn **should have 300 calories**"  
+
+If you think of each of these simple english sentences, the "assertion" piece is the part in bold. In that last example, in other words, after I create a unicorn and it eats three times, I want to assert that its calorie count is **EQUAL TO** 300.  
+
+The assertion library we are going to use today is called [Chai](http://chaijs.com/). Chai can be inserted into many different testing frameworks, but it works seamlessly with Mocha which is what we'll focus on today. Before we get back to our tests, lets look at some of the syntax options Chai provides:  
+
+![Chai Syntax Libraries](http://i.imgur.com/T7Q4YkE.png)  
+
+You'll notice that Chai provides three different interfaces that accomplish the same task - as a developer you can choose which version feels best to you. For today we are going to stick with the [Assert API](http://chaijs.com/api/assert/), since it might seem the most familiar coming off of mythical creatures in mod 1.  
+
+The [Assert API](http://chaijs.com/api/assert/) from Chai provides a plethora of methods to allow for in-depth, dynamic testing. The general syntax looks something like this:  
```

Comment:
This is great. We also try to get them comfortable using "expect" since testing React with Enzyme/Jest uses "expect" for their assertions. I'm happy to not address that in this lesson, though.